TracyOpenCL.hpp

```diff
@@ -139,23 +148,33 @@ namespace tracy {
             }
 #endif
 
-            while (m_tail != m_head)
+            for (; m_tail != m_head; m_tail = (m_tail + 1) % QueryCount)
```

Comment:
This is not for compactness at its root. Previously, the loop body [stops processing on error ](https://github.com/cloudhan/tracy/blob/b0fc0d5dcc921e3647763c9e9f7c333dd922885f/TracyOpenCL.hpp#L149). I changed it for skipping the error events. So the `return` is replaced by a `continue`. It is for correctness here.

Reply:
I understand now, but stopping the processing of events was by design in the case where one event, like a kernel, takes a long time to run. This could potentially halt TracyCLCollect which, in my opinion, is an undesired behaviour. Think about applications that perform the collect at the end of a frame or periodically

Reply:
@wolfpld do you have any thoughts on this one?

Reply:
I am busy recently, will address it later.

Reply:
> @wolfpld do you have any thoughts on this one?

It may be worthwhile to extract the loop body into a separate function.

Reply:
> > @wolfpld do you have any thoughts on this one?
> 
> It may be worthwhile to extract the loop body into a separate function.

Sure, that could be done, but I meant your thoughts on waiting for an event to complete in the collect function. Other APIs ([Vulkan](https://github.com/wolfpld/tracy/blob/master/TracyVulkan.hpp#L237), [OpenGL](https://github.com/wolfpld/tracy/blob/master/TracyOpenGL.hpp#L160)) simply return if an event is not ready to be queried

Reply:
The impact of profiling should be as minimal as it can be. Is this code blocking? Then it has to be changed.

Reply:
After reading what I wrote again I think I wasn't clear enough. Basically the existing code is not blocking, i.e. if an event in the queue is not ready it simply returns to try again in the next collect call, but the change proposed in this PR has the potential to make it blocking, hence my push back